language/puppet-functions.md

```diff
@@ -51,29 +51,60 @@ possible to call a puppet function with a lambda, and possibly also type a param
 
 Function Definition - Scope & Autoloading
 ---
-Puppet Functions are autoloaded from modules. They are located under <module-root>/functions.
-The filename must match the simple (non namespaced name part) of the function. Thus, a function 'min' in module 'testmodule' is placed like this:
+
+#### Autoloading
+
+Puppet Functions are autoloaded from modules and the environment when located under <module-root>/functions or <environment-root>/functions respectively.
+
+The filename must match the simple (non namespaced name part) of the function. Thus, a function 'min' in module 'testmodule' is located like this:
```

Comment:
> ...a function "testmodule::min" in module "testmodule" is located like...

Let's please never truncate namespaces when explaining this stuff. I know it probably seems perfectly clear to you, but it confuses readers.